PacketDotNet/TransportPacket.cs

```diff
@@ -20,6 +21,14 @@ namespace PacketDotNet
     /// </summary>
     public abstract class TransportPacket : Packet
     {
+        /// <summary>
+        /// Callback function for UdpPacket Payload decoding.
+        /// First parameter is the payload, second parameter if the UdpPacket itself
+        /// returned value is the decoded payload as Packet or null if payload is invalid or not supported
+        /// then internal decoding will continue
+        /// </summary>
+        public static Func<ByteArraySegment, TransportPacket, Packet> CustomPayloadDecoder;
```

Comment:
Can you change the result to a ```PacketOrByteArraySegment```? Then we'd also be able to close #103 as well as it'd be possible to return the payload itself.

Reply:
done